### PR TITLE
Add Notification page for receiving scheduled notifications

### PR DESCRIPTION
**Pull request by BatDroid**
✅ I have read the [contributing file](https://github.com/NMF-earth/nmf-app/blob/master/contributing.md)

## Summary
related to #213 . Adding a notification page for receiving scheduled notifications.
I wasn't sure if I should use google translate for translation in other locales.
as it is explained [here](https://github.com/NMF-earth/nmf-app/issues/213#issuecomment-716173650), this page is only available on ios devices.

## Changelog
New notification page available in settings page in ios devices.
By turning on and giving permission, user will receive notification on every sunday 9 pm.

### CODE REVIEW

File: app/screens/Notifications/NotificationsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import * as Notifications from "expo-notifications";
+import { useDispatch, useSelector } from "react-redux";
+
+import { Text } from "../../components";
+import { t } from "../../utils";
+import { userPreferences } from "../../ducks";
+import navigationOptions from "./NotificationsScreen.navigationOptions";
+import styles from "./NotificationsScreen.styles";
+
+const MyLocationScreen = () => {
```
Comment (PierreBresson) [inline app/screens/Notifications/NotificationsScreen.tsx:12]: `const NotificationsScreen = () => {` 😉

File: app/screens/Settings/SettingsScreen.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import {
   TouchableWithoutFeedback,
   ScrollView,
   Image,
+  Platform,
```
Comment (PierreBresson) [inline app/screens/Settings/SettingsScreen.tsx:7]: We have some helper for platform : 
import { t, platform } from "../../utils";

File: app/screens/Settings/SettingsScreen.tsx [hunk 3]
```diff
@@ -60,9 +66,12 @@ const SettingsScreen = (props) => {
 
   return (
     <ScrollView style={styles.container}>
-      {rowItems.map((item, index) => (
-        <SettingsRow key={index} onPress={item.onPress} title={item.title} />
-      ))}
+      {rowItems.map((item, index) => {
+        if (item.os && item.os !== Platform.OS) return null;
```
Comment (PierreBresson) [inline app/screens/Settings/SettingsScreen.tsx:70]:
```
if(...) {
   return null;
}
```

**Review by PierreBresson — CHANGES_REQUESTED** [review]
Almost there :)

**Comment by PierreBresson** [comment]
I've been investigating around android and it seems that it has been done in this [PR](https://github.com/expo/expo/pull/9973)

You can see in [expo/package/expo-notifications](https://github.com/expo/expo/tree/master/packages/expo-notifications) that `WeeklyNotificationTrigger` looks like what we want for Android

So I did a little experiment with `WeeklyNotificationTrigger`, so you can add `"expo-notifications": "https://gitpkg.now.sh/PierreBresson/packages/expo-notifications?main",` instead of `"expo-notifications": "0.7.2"` in `package.json`. Unfortunately, it doesn't work on my android so far...

**Comment by BatDroid** [comment]
I'll add the changes and check the pr that you mentioned in the following days

File: app/screens/Notifications/NotificationsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const dispatch = useDispatch();
+  const [activated, setActivated] = useState(
+    useSelector(userPreferences.selectors.getActivateNotifications)
+  );
+
+  const onValueChange = useCallback(
+    async (value: boolean) => {
+      await Notifications.requestPermissionsAsync();
```
Comment (PierreBresson) [inline app/screens/Notifications/NotificationsScreen.tsx:20]: Also maybe good praticse to wrap await in a try catch

File: app/screens/Settings/SettingsScreen.tsx (hunk 2 above)
Comment (BatDroid) [reply app/screens/Settings/SettingsScreen.tsx:7]: it has just two booleans of isAndroid & isIOS and it doesn't help this case. in this case we want to have Items that are os specific and later check that against it. so if you don't mind I won't apply this change

File: app/screens/Settings/SettingsScreen.tsx (hunk 3 above)
Comment (BatDroid) [reply app/screens/Settings/SettingsScreen.tsx:70]: I wonder why the eslint is not active for this. in my personal projects it is activated by default preventing me to do such a thing.

**Comment by BatDroid** [comment]
> @BatDroid Thanks for your PR! I've been investigating around android weekly notif and it seems that it has been done in this [PR](https://github.com/expo/expo/pull/9973)
> 
> You can see in [expo/package/expo-notifications](https://github.com/expo/expo/tree/master/packages/expo-notifications) that `WeeklyNotificationTrigger` looks like what we want for Android
> 
> So I did a little experiment with `WeeklyNotificationTrigger`, so you can add `"expo-notifications": "https://gitpkg.now.sh/PierreBresson/packages/expo-notifications?main",` instead of `"expo-notifications": "0.7.2"` in `package.json`. Unfortunately, it doesn't work on my android so far...

didn't work on my android either. 
here is my code:
Screen Shot 1399-08-07 at 00 44 24
and here is the catch running it:
Screen Shot 1399-08-07 at 00 44 35

I believe it's better we wait till it is officially merged/tested & released.

**Review by PierreBresson — APPROVED** [review]

**Comment by PierreBresson** [comment]
@BatDroid Good job, thanks a lot for the PR 🤗 
Feel free to send my your address at pierre @ bresson . io if you want some stickers!

**Pull request merged**